clippy_lints/src/unit_like_struct_brackets.rs

```diff
@@ -0,0 +1,53 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use rustc_ast::ast::*;
+use rustc_errors::Applicability;
+use rustc_lint::{EarlyContext, EarlyLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Finds structs without fields ("unit-like structs") that are declared with brackets.
+    ///
+    /// ### Why is this bad?
+    /// Empty brackets after a struct declaration can be omitted.
+    ///
+    /// ### Example
+    /// ```rust
+    /// struct Cookie {}
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// struct Cookie;
+    /// ```
+    #[clippy::version = "1.61.0"]
+    pub UNIT_LIKE_STRUCT_BRACKETS,
+    style,
+    "finds struct declarations with empty brackets"
+}
+declare_lint_pass!(UnitLikeStructBrackets => [UNIT_LIKE_STRUCT_BRACKETS]);
+
+impl EarlyLintPass for UnitLikeStructBrackets {
+    fn check_item(&mut self, cx: &EarlyContext<'_>, item: &Item) {
+        if let ItemKind::Struct(var_data, _) = &item.kind && has_no_fields(var_data) {
+            let span_after_ident = item.span.with_lo(item.ident.span.hi());
+
+            span_lint_and_sugg(
```

Comment:
The help message reads `help: remove the brackets: `;``. 
I really dislike the semicolon there. Is there another function that does the same, but does not print the suggestion?

Reply:
You could use `span_lint_and_then` with [`span_suggestion_hidden`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_errors/diagnostic/struct.Diagnostic.html#method.span_suggestion_hidden)

Reply:
Perfect!